ir/value.cpp

```diff
@@ -188,22 +173,24 @@ StateValue Input::toSMT(State &s) const {
   expr type = getTyVar();
 
   expr val;
-  if (attributes & ByVal) {
+  if (hasAttribute(ParamAttrs::ByVal)) {
     unsigned bid;
     string sz_name = getName() + "#size";
     expr size = expr::mkVar(sz_name.c_str(), bits_size_t-1).zext(1);
     val = get_global(s, getName(), size, 1, false, bid);
     s.getMemory().markByVal(bid);
   } else {
-    val = getType().mkInput(s, smt_name.c_str(), attributes);
+    val = getType().mkInput(s, smt_name.c_str(), attrs);
   }
 
   if (!config::disable_undef_input) {
-    auto [undef, vars] = getType().mkUndefInput(s, attributes);
-    for (auto &v : vars) {
-      s.addUndefVar(move(v));
+    auto [undef, vars] = getType().mkUndefInput(s, attrs);
+    if (undef.isValid()) {
```

Comment:
why this change? this is unrelated.
in general, we don't optimize for !isValid() case since most formulas should be valid.

Reply:
Oh yes, this is an unrelated change. I'll revert this change.